### PR TITLE
Improve travis deploy

### PR DESCRIPTION
**Pull request by webknjaz**
Here's my improvement proposal: scripting you do inside of `deploy` stage is already done by deployment provider, so why not reuse it?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,28 +24,34 @@ script:
...
-              twine upload dist/* --username andrew.svetlov --password $PYPI_PASSWD;
-          fi
-      after_success: true  # do nothing
+      deploy:
+        provider: pypi
+        on:
+          tags: true
+        user: andrew.svetlov
+        password: $PYPI_PASSWD
+        distributions: sdist bdist_wheel
```
Comment (webknjaz) [inline .travis.yml]: See: https://github.com/travis-ci/dpl/blob/e571131/lib/dpl/provider/pypi.rb#L96-L97.
It already encapsulates twine as well.

File: .travis.yml [hunk 2]
```diff
@@ -24,28 +24,34 @@ script:
...
-              python setup.py bdist_wheel;
-              twine upload dist/* --username andrew.svetlov --password $PYPI_PASSWD;
-          fi
-      after_success: true  # do nothing
+      deploy:
+        provider: pypi
+        on:
+          tags: true
+        user: andrew.svetlov
+        password: $PYPI_PASSWD
```
Comment (webknjaz) [inline .travis.yml]: Just move password here

File: .travis.yml [hunk 3]
```diff
@@ -24,28 +24,34 @@ script:
...
-        - if [ ! -z $TRAVIS_TAG ];then
-              python setup.py sdist;
-              python setup.py bdist_wheel;
-              twine upload dist/* --username andrew.svetlov --password $PYPI_PASSWD;
-          fi
-      after_success: true  # do nothing
+      deploy:
+        provider: pypi
+        on:
+          tags: true
```
Comment (webknjaz) [inline .travis.yml]: This implies `! -z $TRAVIS_TAG` check

**Comment by asvetlov** [comment]
Well, it my first try but I get some strange error like 'password key is not supported'.
Maybe I had a just typo in this name, not sure.

I'll try your proposal again.

On other side proposed structure fits better to projects like aiohttp, yarl and multidict -- we need support manylinux binary builds, thus we should make custom deployment stage.

BTW would you try to convert our small projects: yarl and multidict?
If yes -- I see no problems with giving you commit rights to these repos and register you as writer to appropriate PyPI projects.
You'll be able to test deployments under your credentials.
The only very ask -- don't publish releases but pre-releases only. They are not installed by `pip` by default, your experiments will not hurt our users.

Please let me know if you agree with the proposal.

**Comment by webknjaz** [comment]
I'm okay with trying to convert small projects. I'm currently trying to optimize what I've found in aiohttp and eliminate dead stuff

**Comment by webknjaz** [comment]
Regarding password, it is common to do mistakes while using Travis CLI (I've seen this happening a couple of times), just re-try.
You may set [`PYPI_PASSWORD` secure env var in Travis CI settings (src)](https://github.com/travis-ci/dpl/blob/e571131/lib/dpl/provider/pypi.rb#L12) for testing purposes, but I personally prefer to see it in YAML config just for the case if there's serveral providers enabled needing different credentials.

**Comment by webknjaz** [comment]
Regarding docker builds, I'm aiming to move the build itself to corresponding step since `after_success` doesn't alert you via failed/errored job status if smth went wrong. I think I can decouple those scripts.

**Comment by asvetlov** [comment]
Let's keep discussion about multidict improvements in private channel.

File: .travis.yml [hunk 4]
```diff
@@ -24,28 +24,34 @@ script:
...
 
+_helpers:
+- &_skip_steps
+  python: 3.5
+  before_install: skip
+  install: skip
+  before_script: skip
+  script: skip
+  after_script: skip
+  after_success: skip
```
Comment (webknjaz) [inline .travis.yml]: @asvetlov this needs to be `[]` because of previously discovered bug

File: .travis.yml (hunk 3 above)
Comment (webknjaz) [reply .travis.yml]: @asvetlov new `if :` pre-processing check is better nowadays

File: .travis.yml (hunk 2 above)
Comment (webknjaz) [reply .travis.yml]: @asvetlov this also needs to be updated

File: .travis.yml [hunk 5]
```diff
@@ -24,28 +24,34 @@ script:
...
-          fi
-      after_success: true  # do nothing
+      deploy:
+        provider: pypi
+        on:
+          tags: true
+        user: andrew.svetlov
+        password: $PYPI_PASSWD
+        distributions: sdist bdist_wheel
+        skip_upload_docs: true
```
Comment (webknjaz) [inline .travis.yml]: this is default currently

**Comment by asvetlov** [comment]
@webknjaz would you make a new PR for fixing mentioned notes.

**Comment by webknjaz** [comment]
Yeah, at some point. Just wanted to document this, so it won't get lost.

--Sviatoslav.

**Pull request merged**